UnityProject/Assets/Scripts/Items/Weapons/Explosive.cs

```diff
@@ -0,0 +1,190 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Systems.Explosions;
+using ScriptableObjects.Communications;
+using Communications;
+using Managers;
+using Objects;
+using UI;
+using UnityEngine;
+
+
+namespace Items.Weapons
+{
+	public class Explosive : SignalReceiver, ICheckedInteractable<PositionalHandApply>, ICheckedInteractable<MouseDrop>
+	{
+		[Header("Explosive settings")]
+		[SerializeField] private ExplosiveType explosiveType;
+		[SerializeField] private bool detonateImmediatelyOnSignal;
+		[SerializeField] private int timeToDetonate = 10;
+		[SerializeField] private int minimumTimeToDetonate = 10;
+		[SerializeField] private ExplosionComponent explosionPrefab;
+		[Header("Explosive Components")]
+		[SerializeField] private SpriteHandler spriteHandler;
```

Comment:
why not get these on Awake?

Reply:
idk i just feel like this is the better way to do it, i can change this if you want

Reply:
I think it is better to auto get those, less hassle when doing the prefabs

Reply:
gotcha, will change this once im done with another possible PR im working on

Reply:
Done 👍